### ISSUE TITLE
ExceptionHandler cannot handle tasks properly

### ISSUE THREAD

**Issue by Hau-Hau** [jscarle/LightResults.Extensions#6]
Hi @jscarle and thank you for your work!

We have started using LightResults in our project and we have found that ExceptionHandler is not able to properly handle async methods.
Lets consider below case:
```csharp
    class Program
    {
        public static async Task DoSomething() {
            await Task.Delay(5);
        }

        public static async Task Main()
        {
            Result<Task> result = ExceptionHandler.Try(DoSomething);
        }
    }
```

In such case I would want to get `Task<Result>` rather than `Result<Task>`.
However, it seems to be correct behaviour for `Try` method, because of that I would suggest to implement additional subset of `TryAsync` methods.
Also, we will need to consider Task and TaskValue invariants.
What do you think about that?

--
sample implementation
```csharp
    public static async Task<Result<TResult>> TryAsync<TResult>(Func<Task<TResult>> func)
    {
        try
        {
            return Result<TResult>.Ok(await func());
        }
        catch (Exception ex)
        {
            return HandleException<TResult>(ex);
        }
    }
```

**Comment by Hau-Hau** [comment]
I have implemented changes in https://github.com/jscarle/LightResults.Extensions/pull/7
Covers only Task as library targats netstandard2.0, no need for covering ValueTask.

**Comment by jscarle** [comment]
Hello @Hau-Hau!

Thanks for contributing. I merged your PR and I also added support for ValueTask while also adding all of the extension methods. So you should be able to call TryAsync directly on a Task.

The changes are published as [v8.0.2](https://www.nuget.org/packages/LightResults.Extensions.ExceptionHandling/8.0.2). Let me know if you have any issues.

Cheers.
